api/data_explorer/util/elasticsearch_util.py

```diff
@@ -153,12 +102,95 @@ def _get_bucket_interval(field_range):
         return 1000000000000
 
 
-def get_samples_overview_facet(es_field_names):
-    filters = {
-        facet: Match(**{field: True})
-        for facet, field in es_field_names.iteritems()
-    }
-    return NestedFacet('samples', FiltersFacet(filters))
+def _range_to_number(interval_str):
+    """Converts "X-Y" -> "X"."""
+    if not '-' in interval_str:
+        return int(interval_str)
+
+    number = interval_str.split('-')[0]
+    number = number.replace('M', '000000')
+    number = number.replace('B', '000000000')
+    if '.' in number:
+        return float(number)
+    else:
+        return int(number)
+
+
+def convert_to_index_name(s):
+    """Converts a string to an Elasticsearch index name."""
+    # For Elasticsearch index name restrictions, see
+    # https://github.com/DataBiosphere/data-explorer-indexers/issues/5#issue-308168951
+    # Elasticsearch allows single quote in index names. However, they cause other
+    # problems. For example,
+    # "curl -XDELETE http://localhost:9200/nurse's_health_study" doesn't work.
+    # So also remove single quotes.
+    prohibited_chars = [
+        ' ', '"', '*', '\\', '<', '|', ',', '>', '/', '?', '\''
+    ]
+    for char in prohibited_chars:
+        s = s.replace(char, '_')
+    s = s.lower()
+    # Remove leading underscore.
+    if s.find('_', 0, 1) == 0:
+        s = s.lstrip('_')
+    print('Index name: %s' % s)
+    return s
+
+
+def deserialize(filter_arr, es_facets):
```

Comment:
Since this is moving out of facets_controller, it should have a more descriptive name. `get_facet_value_dict`

Reply:
Done.